alibi/explainers/cfproto.py

```diff
@@ -182,16 +185,15 @@ def __init__(self,
         self.clip = clip
         self.write_dir = write_dir
 
-        # compute dimensionality after conversion from OHE to ordinal encoding
-        shape = ohe_to_ord_shape(shape, cat_vars=cat_vars, is_ohe=self.ohe)
-
         if self.is_cat:
+            # compute dimensionality after conversion from OHE to ordinal encoding
+            shape = ohe_to_ord_shape(shape, cat_vars=cat_vars, is_ohe=self.ohe)
 
             # define ragged tensor for mapping from categorical to numerical values
-            self.map_cat_to_num = tf.ragged.constant([np.zeros(v) for _, v in self.cat_vars.items()])
+            self.map_cat_to_num = tf.ragged.constant([np.zeros(v) for _, v in cat_vars.items()])
 
             # define placeholder for mapping which can be fed after the fit step
-            max_key = max(cat_vars, key=cat_vars.get)
+            max_key = max(cat_vars)
```

Comment:
`max` is by default over the keys. This also avoids a type error because `dict.get` actually returns an `Optional` type.

Reply:
#610 
De default is not over the keys. Both functions generate different outputs.

cat_vars
{0: 2, 2: 3, 5: 8, 13: 2, 15: 2, 17: 2, 19: 3}
max(cat_vars)
19
max(cat_vars,key=cat_vars.get)
5